### PR TITLE
Build Improvements

### PR DESCRIPTION
**Pull request by apachont**
- Fixed absolute paths for Checkstyle, sql-parser and OpenApi
- Updated Checkstyle/SuppressionFilter and Checkstyle/Header modules 
- Updated javacc from version 4.0.1 to 4.1.3
- Disable timestamp autogeneration on Swagger

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -99,9 +99,9 @@ allprojects {
 
     //Temporary hack to disable caching of Test tasks.
     //https://github.com/gradle/gradle/issues/9210
-    /*tasks.withType(Test).configureEach {
+    tasks.withType(Test).configureEach {
         outputs.upToDateWhen { false }
```
Comment (jprinet) [inline build.gradle:103]: I would prefer `doNotTrackState` as you can provide a reason for disabling optimizations https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:disable-state-tracking

File: buildscripts/sql-parser-generator.gradle [hunk 2]
```diff
@@ -60,11 +60,8 @@ task extractParserTemplate(type: Copy) {
 }
 
 task generateFmppSources() {
-    inputs.files(layout.buildDirectory.dir('codegen/templates'))
-        .withPropertyName('sourceFiles')
-        .withPathSensitivity(PathSensitivity.RELATIVE)
-    outputs.dir(layout.buildDirectory.dir('generated-sources/fmpp'))
-        .withPropertyName('outputDir')
+    inputs.files(layout.buildDirectory.dir('codegen/templates')).withPropertyName('sourceFiles').withPathSensitivity(PathSensitivity.RELATIVE)
```
Comment (jprinet) [inline buildscripts/sql-parser-generator.gradle:63]: If there is no code change, I don't see a good reason to reformat this line

File: buildscripts/sql-parser-generator.gradle [hunk 3]
```diff
@@ -60,11 +60,8 @@ task extractParserTemplate(type: Copy) {
...
+    outputs.dir(layout.buildDirectory.dir('generated-sources/fmpp')).withPropertyName('outputDir')
```
Comment (jprinet) [inline buildscripts/sql-parser-generator.gradle:64]: same as above

File: modules/api/src/test/java/org/apache/ignite/lang/IgniteExceptionArchTest.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ public Set<Location> get(Class<?> testClass) {
     }
 
     @SuppressWarnings("unused")
-    //@ArchTest
+    @ArchTest
```
Comment (jprinet) [inline modules/api/src/test/java/org/apache/ignite/lang/IgniteExceptionArchTest.java:55]: Is this related to build optimization?

File: modules/client/src/test/java/org/apache/ignite/client/ConfigurationTest.java [hunk 5]
```diff
@@ -31,13 +31,11 @@
 import java.util.concurrent.Executors;
 import org.apache.ignite.Ignite;
 import org.apache.ignite.lang.IgniteException;
-import org.junit.jupiter.api.Disabled;
```
Comment (jprinet) [inline modules/client/src/test/java/org/apache/ignite/client/ConfigurationTest.java:34]: If a test was disabled there is no reason to re-enable it

File: modules/client/src/test/java/org/apache/ignite/client/RetryPolicyTest.java [hunk 6]
```diff
@@ -39,13 +39,11 @@
 import org.apache.ignite.table.Tuple;
 import org.apache.ignite.tx.Transaction;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Disabled;
```
Comment (jprinet) [inline modules/client/src/test/java/org/apache/ignite/client/RetryPolicyTest.java:42]: same as above

File: settings.gradle [hunk 7]
```diff
@@ -15,12 +15,7 @@
  * limitations under the License.
  */
 
-plugins {
```
Comment (jprinet) [inline settings.gradle:18]: I guess your main branch state is not the same as upstream, please restore the state to clarify this PR

**Review by jprinet — CHANGES_REQUESTED** [review]
I guess your `main` branch state is not the same as `upstream`, please restore the state to clarify this PR.

For instance:
- Some tests are disabled
- I don't see any CommandLineArgumentProvider

**Comment by apachont** [comment]
Moved to #2
